oak/common/app_config.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/common/app_config.h"
+
+#include <set>
+#include <utility>
+
+#include "asylo/util/logging.h"
+
+namespace oak {
+
+bool ValidApplicationConfig(ApplicationConfiguration* config) {
+  if (config == nullptr) {
+    return false;
+  }
+  // Check name uniqueness and count pseudo-nodes.
+  using fqpn = std::pair<std::string, std::string>;
```

Comment:
Could you clarify what the components mean?

Reply:
Done.